tests/NCronJob.Tests/RetryTests.cs

```diff
@@ -0,0 +1,237 @@
+using System;
+using System.Collections.Specialized;
+using System.Globalization;
+using System.Threading.Channels;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Polly;
+using Shouldly;
+
+namespace NCronJob.Tests;
+
+public sealed class RetryTests : JobIntegrationBase
```

Comment:
@linkdotnet FWIW, This test suite has been mostly rewritten. Please review it with a pair of critical eyes.

Reply:
I checked the old and the new version of you.
If I am not mistaken the gist is the same - of course your tests are asserting more internal bits than the tests before.

Reply:
>If I am not mistaken the gist is the same

Yes it should (Hopefully, if I didn't make any mistake).

One notable addition is the introduction of a long running job that gets cancelled mid-flight (by both the CancelJobs() and the shutdown of the host).